service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpSendHandler.java

```diff
@@ -266,7 +266,15 @@ public void onDelivery(Event event)
             {
                 //By closing the link locally, proton-j will fire an event onLinkLocalClose. Within ErrorLoggingBaseHandlerWithCleanup,
                 // onLinkLocalClose closes the session locally and eventually the connection and reactor
-                log.debug("Closing amqp cloud to device message sender link since the message was delivered");
+                if (remoteState.getClass().equals(Accepted.class))
+                {
+                    log.debug("Closing AMQP cloud to device message sender link since the message was delivered");
+                }
+                else
+                {
+                    log.debug("Closing AMQP cloud to device message sender link since the message failed to be delivered");
+                }
+
```

Comment:
This change is completely orthogonal to the error code work that this PR is for, but it is a small change that is worth fixing despite that